### PR TITLE
New Flare Integration API

### LINKED ISSUE #219: about flare'controller mechanism
Hi, I saw the "flare" example, now I wan't to play audio when the flare animation reach some keyframe, In the flare runtime, there is a parameter "controller" where I can call the api: triggerEvent to test if there is a event, but in the flame's FlareAnimation class, I can't find any interface where i can do that.    Is this not support or something else i missed?

if there is no such mechanism to do that, is there any recommend for my propose (play sound when the flare's keyfame reach )?

Waiting for your reply, thanks very much!

### PR DESCRIPTION
**Pull request by renancaraujo**
Api preview:

```dart


class MySuperDuperFlareComponent extends FlareActorComponent {
  FlareControls
      flareControls; // FlareControls is the default implementation of FlareController
  MySuperDuperFlareComponent(this.flareControls)
      : super(
          "assets/my_flare.flr",
          controller: flareControls,
          fit: BoxFit.cover,
        ) {
    width = 200;
    height = 500;
  }

  playThisCoolAnimation() {
    flareControls.play("A cool animation");
  }
}


```

# Description

Add a new API for using [flare](https://github.com/2d-inc/Flare-Flutter) on flame. 

Fixes #219

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on `1.0`
- [x] This PR is targeted to merge into `1.0` (not `master`)
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] I have added examples for new features in `doc/examples`
- [ ] The continuous integration (CI) is passing

### CODE REVIEW

**Comment by erickzanardo** [comment]
Rebase to the develop branch please

File: lib/flare/flare_flame.dart [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  final bool shouldClip;
+  final FlareController controller;
+  final FlareCompletedCallback callback; //
+  final Color color; //
+  final String boundsNode; //
+  final bool sizeFromArtboard; //
+
+
+
+
```
Comment (erickzanardo) [inline lib/flare/flare_flame.dart:64]: Too many blank lines

Comment (renancaraujo) [reply lib/flare/flare_flame.dart:64]: still draft tho

**Comment by erickzanardo** [comment]
This solves #219

File: lib/flare/flare_flame.dart [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+import 'package:flare_flutter/flare_controller.dart';
+import 'package:flare_flutter/provider/asset_flare.dart';
+
+import 'package:flare_dart/math/aabb.dart';
+import 'package:flare_dart/math/mat2d.dart';
+import 'package:flare_dart/math/vec2d.dart';
+
+class FlareActorPipelineOwner extends PipelineOwner {}
+
+class FlareActorComponent extends PositionComponent {
```
Comment (spydon) [inline lib/flare/flare_flame.dart:20]: The file should maybe be named something that reflects this?

Comment (renancaraujo) [reply lib/flare/flare_flame.dart:20]: Wilco

**Comment by erickzanardo** [comment]
Is this still in WIP? Change the title if not, and point your PR to the v1.0.0 branch instead of master

File: doc/examples/flare/lib/main.dart [hunk 3]
```diff
@@ -15,90 +18,97 @@ void main() {
...
+  MinionComponent minionComponent;
+  MinionController minionController;
 
-  FlareAnimation flareAnimation;
   bool loaded = false;
 
   MyGame() {
-    _start();
+    minionController = MinionController();
+    minionComponent = MinionComponent(minionController);
```
Comment (erickzanardo) [inline doc/examples/flare/lib/main.dart:34]: Suggestion, cascade notation here

**Comment by erickzanardo** [comment]
@renancaraujo I think it is safe to close this right? you have already moved this changes to the other repositories?

**Comment by renancaraujo** [comment]
Thi shall be tackled via: https://github.com/flame-engine/flame-flare/pull/2

**Pull request closed without merging**